Base_DotNet.Dockerfile

```diff
@@ -7,24 +7,31 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ENV DOTNET_ROOT=/usr/dotnet
 ENV PATH="$DOTNET_ROOT:${PATH}"
 
-RUN apt-get -q update \
- && DEBIAN_FRONTEND=noninteractive apt-get -q install -y --no-install-recommends \
+RUN apt-get --quiet update \
+ && export DEBIAN_FRONTEND=noninteractive \
+ && apt-get --quiet --yes upgrade \
+ && apt-get --quiet --yes --no-install-recommends install \
     curl \
+    gnupg \
     libicu-dev \
-
+ && echo '----- Azure Functions Core Tools' \
+ && curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | tee /etc/apt/trusted.gpg.d/microsoft.gpg >/dev/null\
+ && curl https://packages.microsoft.com/config/debian/11/prod.list | tee /etc/apt/sources.list.d/microsoft-prod.list >/dev/null\
+ && apt-get update \
+ && apt-get --quiet --yes --no-install-recommends install \
+    azure-functions-core-tools-4 \
  && echo '----- DotNet Core' \
  && mkdir -p "$DOTNET_ROOT" \
  && curl -sL https://dot.net/v1/dotnet-install.sh -o "$DOTNET_ROOT/dotnet-install.sh" \
  && chmod u+x "$DOTNET_ROOT/dotnet-install.sh" \
  && "$DOTNET_ROOT/dotnet-install.sh" --install-dir "$DOTNET_ROOT" -c current \
-

```

Comment:
revert

Reply:
IntelliJ was showing it as invalid, since the preceding `\` would escape one newline, but not both. So I could either remove the line or add a `\`